pygmt/src/grdcontour.py

```diff
@@ -96,32 +102,57 @@ def grdcontour(self, grid, **kwargs):
     Example
     -------
     >>> import pygmt
-    >>> # load the 15 arc-minutes grid with "gridline" registration
-    >>> # in a specified region
+    >>> # Load the 15 arc-minutes grid with "gridline" registration in the
+    >>> # specified region
     >>> grid = pygmt.datasets.load_earth_relief(
     ...     resolution="15m",
     ...     region=[-92.5, -82.5, -3, 7],
     ...     registration="gridline",
     ... )
-    >>> # create a new plot with pygmt.Figure()
+    >>> # Create a new plot with pygmt.Figure()
     >>> fig = pygmt.Figure()
-    >>> # create the contour plot
+    >>> # Create the contour plot
     >>> fig.grdcontour(
-    ...     # pass in the grid downloaded above
+    ...     # Pass in the grid downloaded above
     ...     grid=grid,
-    ...     # set the interval for contour lines at 250 meters
+    ...     # Set the interval for contour lines at 250 meters
     ...     interval=250,
-    ...     # set the interval for annotated contour lines at 1,000 meters
+    ...     # Set the interval for annotated contour lines at 1,000 meters
     ...     annotation=1000,
-    ...     # add a frame for the plot
+    ...     # Add a frame for the plot
     ...     frame="a",
-    ...     # set the projection to Mercator for the 10 cm figure
+    ...     # Set the projection to Mercator for the 10 cm figure
     ...     projection="M10c",
     ... )
-    >>> # show the plot
+    >>> # Show the plot
     >>> fig.show()
     """
     kwargs = self._preprocess(**kwargs)
+
+    # Backward compatibility with the old syntax like (100, "e", "f10p", "gred").
```

Comment:
Still need to add a test to increase the code coverage. Instead of writing a new test with a new baseline image, you can create a new test based on `test_grdcontour_one_level`.

I think the one below should produce an image that is identical to `test_grdcontour_one_level.png`:
```
    fig = Figure()
    fig.grdcontour(
        grid=grid, interval=[400], annotation=[570, "pblack"], projection="M10c", frame=True
    )
```

Reply:
This test is added in commit https://github.com/GenericMappingTools/pygmt/pull/3116/commits/a05ec74ec721dc56bb6cf2f07d2627e60247f632.